app/scripts/controllers/transactions.js

```diff
@@ -251,10 +252,17 @@ angular.module('openhimConsoleApp')
       // date filter
       filterDateStart = $scope.settings.filter.startDate;
       filterDateEnd = $scope.settings.filter.endDate;
-      if(filterDateStart && filterDateEnd) {
-        var startDate = moment(filterDateStart).format();
-        var endDate = moment(filterDateEnd).endOf('day').format();
-        filtersObject.filters['request.timestamp'] = JSON.stringify( { '$gte': startDate, '$lte': endDate } );
+  
+      if(filterDateStart || filterDateEnd) {
+        var buildDateFilterObject = function(object, comparator, date) {
```

Comment:
Maybe move this function of the parent function. It gets created every time the 'returnFilters' function is run.